### ISSUE TITLE
Wishlist: disable the whiteboard during the tutorial (GNA #19223)

### ISSUE THREAD

**Issue by wesnoth-bugs** [wesnoth/wesnoth#1285]
**[Original submission](https://gna.org/bugs/index.php?19223)** by octalot on 2011-12-28

An accidental keypress can enable the whiteboard during the tutorial, but the whiteboard is a more complex concept than most of the things in the tutorial.

I'd like to wishlist disabling the whiteboard during the tutorial.

(Reproduced on Debian Unstable)
Release: 1.9.13
Priority: 3 - Low
Severity: 1 - Wish

**Comment by wesnoth-bugs** [comment]
**Modified on 2012-04-22**

**gabba wrote:**

> This is needed, but detecting whether the current game is the tutorial without an ugly hack isn't as easy as it sounds.
> 
> 
> 
> Will speak with other devs and hopefully find a clean solution.

**gabba changed status:** None -> Confirmed

**gabba changed priority:** 5 - Normal -> 3 - Low

**Comment by wesnoth-bugs** [comment]
**Modified on 2012-05-03**

**spixi wrote:**

> Maybe you could introduce a ScenarioWML option  disable_planning_mode, which sets a flag which is saved in an additional attribute of play_controller and lets wb::manager::can_activate() return false.
> 
> 
> 
> It would be good, if this option could be used in [tutorial], [test], [scenario] and [multiplayer] tags.

**Comment by jostephd** [comment]
Nowadays when whiteboard is enabled there is a message across the mainmap that says, among other things, "Press p to disable". This message didn't exist when this bug was filed.

**Comment by CelticMinstrel** [comment]
If disabling the whiteboard is still desired, probably the easiest way is to add Lua/WML API to do it. Not sure if that's strictly a good idea, but at least it's less of a hack than somehow trying to detect whether the current game is the tutorial.

**Comment by gfgtdf** [comment]
I think it'd rather easy to check whether the current game is the tutorial, and the API solution would have the disadvantage that it could not be backported to 1.14

**Comment by jostephd** [comment]
Detecting whether the current game should be easy because it uses a [tutorial] tag, not a [campaign] tag.

**Comment by CelticMinstrel** [comment]
I was actually thinking that too... I wonder if that wasn't the case when this issue was originally filed? But is the game type remembered through the entire game session or only until the scenario starts? I can't remember how that works...

**Comment by jostephd** [comment]
> But is the game type remembered through the entire game session or only until the scenario starts? I can't remember how that works...

That's not important. Even if currently we discard the information early we can just hold on to it for a little longer.

I'm more worried about not introducing arbitrary differences between the tutorial and "real" campaigns. In real wesnoth play the whiteboard is a thing, so I wouldn't want to disable it completely. I would prefer to have the advisor (Delfador in S1, or the side=3 elf in S2) say something like:

> $student, you enabled _planning mode_, which changes the way you interact with Wesnoth. Planning mode is outside the scope of this tutorial, so please press _p_ to disable it again. You can learn more about planning mode in the _Gameplay_ section of the in-game help (press _F1_ to open it now).

**Comment by soliton-** [comment]
I agree that a note is preferable to arbitrarily disabling planning mode.

**Comment by CelticMinstrel** [comment]
Is it even possible to have WML events reacting to planning mode? If I'm not mistaken, your proposal requires adding a whole new unsynced event to the engine.

**Comment by stevecotton** [comment]
> Nowadays when whiteboard is enabled there is a message across the mainmap that says, among other things, "Press p to disable". This message didn't exist when this bug was filed.

As the original logger of the bug, I agree. However, the message only appears for a short duration, to solve this I think it would need to stay permanently visible.

**Comment by CelticMinstrel** [comment]
There should probably also be some very clear indicator in the UI that the whiteboard is enabled.

**Comment by stevecotton** [comment]
> There should probably also be some very clear indicator in the UI that the whiteboard is enabled.

Sounds like a usecase for #5837 to show a message at the top or bottom of the screen.

**Comment by Wedge009** [comment]
This particular tutorial is now part of the classic edition of HttT (HttTC). But given the concern is about avoiding a new player from accidentally entering advanced modes with a hot-key, this seems complicated by the philosophy of the current tutorial of The South Guard, which is supposed to be accessible for both new and experienced players.

It sounds like the consensus is, instead, to have a permanent indication of planning mode being on/off. I'll rename this accordingly.

(For a separate issue, I recently looked at something across 1.10, 1.12, 1.14, 1.16, and 1.18 in quick succession. I can tell you even having the temporary notice that planning mode is enabled/disabled is better than nothing!)

**Comment by Wedge009** [comment]
I belatedly realised the specific request about having a planning mode indicator was logged in #5890. I'm inclined to close this, then, as it was originally pertaining to disabling planning mode in the tutorial altogether. If people genuinely think planning mode should be disabled in The South Guard, then this can be re-opened.
